src/core/main.c

```diff
@@ -1982,6 +1983,12 @@ static int do_reexecute(
                         log_error_errno(r, "Failed to switch root, trying to continue: %m");
         }
 
+        if (getpid_cached() != 1) {
+                r = capability_ambient_set_apply(capability_ambient_set, /* also_inherit= */ false);
+                if (r < 0)
+                        return log_error_errno(errno, "Failed to apply the starting ambient set: %m");
```

Comment:
`r`, not `errno`

Reply:
is it `r` or `-r` if the function returns `-errno`?

Reply:
always `r`